js/ui/selection/selection.strategy.deferred.js

```diff
@@ -179,37 +179,39 @@ export default SelectionStrategy.inherit({
         });
     },
 
-    _addSelectionFilter: function(isDeselect, filter, isSelectAll) {
+    _addSelectionFilter: function(isDeselect, filter, isSelectAll, skipFilter) {
         const that = this;
         const currentFilter = isDeselect ? ['!', filter] : filter;
         const currentOperation = isDeselect ? 'and' : 'or';
         let needAddFilter = true;
         let selectionFilter = that.options.selectionFilter || [];
 
         selectionFilter = that._denormalizeFilter(selectionFilter);
+        if(selectionFilter && selectionFilter.length && !skipFilter) {
```

Comment:
Minor comment.

Let's make this condition shorter:
```suggestion
        if(selectionFilter?.length && !skipFilter) {
```